### PR TITLE
3163 inbound return details view

### LINKED ISSUE #3163: Inbound Returns Detail View
## Is your feature request related to a problem? Please describe 👀

Inverse of Outbound Returns detail view, or much like inbound shipment.

Start with skeleton

### PR DESCRIPTION
**Pull request by lache-melvin**
Fixes #3163

# 👩🏻‍💻 What does this PR do? 
 

Adds detail view for Inbound Return:
Screenshot 2024-03-07 at 4 49 01 PM

Does not include the add button/edit on row click, as I think this modal needs some more work that should be covered off in a separate PR (this one is already big, sorry!)

### CODE REVIEW

File: client/packages/invoices/src/Returns/InboundDetailView/DetailView.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import { SidePanel } from './SidePanel/SidePanel';
+import { ActivityLogList } from 'packages/system/src';
+import { Footer } from './Footer';
+
+export const InboundReturnDetailView: FC = () => {
+  const { data, isLoading } = useReturns.document.inboundReturn();
+  const t = useTranslation('distribution');
+  const navigate = useNavigate();
+
+  const onRowClick = () => {};
```
Comment (CarlosNZ) [inline client/packages/invoices/src/Returns/InboundDetailView/DetailView.tsx:27]: Maybe just chuck a "TO-DO" comment in here so we notice it later on?

File: client/packages/invoices/src/Returns/InboundDetailView/Footer/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import { OnHoldButton } from './OnHoldButton';
+
+const createStatusLog = (invoice: InboundReturnFragment) => {
+  const statusIdx = outboundStatuses.findIndex(s => invoice.status === s);
+  const statusLog: Record<InvoiceNodeStatus, null | undefined | string> = {
+    [InvoiceNodeStatus.New]: null,
+    [InvoiceNodeStatus.Picked]: null,
+    [InvoiceNodeStatus.Shipped]: null,
+    [InvoiceNodeStatus.Delivered]: null,
+    [InvoiceNodeStatus.Verified]: null,
```
Comment (CarlosNZ) [inline client/packages/invoices/src/Returns/InboundDetailView/Footer/Footer.tsx:28]: I don't think we want "Picked" or "Shipped" for Inbounds, do we?

File: client/packages/invoices/src/Returns/InboundDetailView/Footer/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    statusLog[InvoiceNodeStatus.Shipped] = invoice.shippedDatetime;
+  }
+  if (statusIdx >= 4) {
+    statusLog[InvoiceNodeStatus.Delivered] = invoice.deliveredDatetime;
+  }
+  if (statusIdx >= 5) {
+    statusLog[InvoiceNodeStatus.Verified] = invoice.verifiedDatetime;
+  }
+  return statusLog;
+};
```
Comment (CarlosNZ) [inline client/packages/invoices/src/Returns/InboundDetailView/Footer/Footer.tsx:48]: Yup, definitely need to pull this out into common methods at some point ;)

File: client/packages/invoices/src/Returns/InboundDetailView/Footer/Footer.tsx (hunk 2 above)
Comment (CarlosNZ) [reply client/packages/invoices/src/Returns/InboundDetailView/Footer/Footer.tsx:28]: Maybe shipped, but probably not Picked?

File: client/packages/invoices/src/Returns/InboundDetailView/SidePanel/AdditionalInfoSection.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            />
+          </PanelField>
+        </PanelRow> */}
+
+        {/* <PanelLabel>{t('heading.comment')}</PanelLabel>
+        <BufferedTextArea
+          disabled={isDisabled}
+          onChange={e => update({ comment: e.target.value })}
+          value={comment || ''}
+        /> */}
```
Comment (CarlosNZ) [inline client/packages/invoices/src/Returns/InboundDetailView/SidePanel/AdditionalInfoSection.tsx:49]: Cool -- will comment out the same stuff in Outbounds

File: client/packages/invoices/src/utils.ts [hunk 5]
```diff
@@ -156,6 +174,18 @@ export const isInboundDisabled = (inbound: InboundRowFragment): boolean => {
         inbound.status === InvoiceNodeStatus.Verified;
 };
 
+export const isInboundReturnDisabled = (
+  inboundReturn: InboundReturnRowFragment
+): boolean => {
+  // const isManuallyCreated = !inbound.linkedShipment?.id;
```
Comment (CarlosNZ) [inline client/packages/invoices/src/utils.ts:180]: Do we have an issue to handle the auto-linked invoices?

**Review by CarlosNZ — APPROVED** [review]
LGTM, thanks. Lot of copy-pasting there, I know it's annoying ;)

Couple of comments/questions, but nothing really actionable.

Cheers

File: client/packages/invoices/src/Returns/InboundDetailView/Footer/Footer.tsx (hunk 2 above)
Comment (lache-melvin) [reply client/packages/invoices/src/Returns/InboundDetailView/Footer/Footer.tsx:28]: Went with this as that's what we have for Inbound Shipments (I think we pass through the Picked status from connected Outbound so we can see progress??)

Though after the conversation about maybe there being no `Picked` status on the Outbound Return... might wait for confirmation from Richard on what these will be?

File: client/packages/invoices/src/utils.ts (hunk 5 above)
Comment (lache-melvin) [reply client/packages/invoices/src/utils.ts:180]: We have one for "creating the other half with processors" but I'll add to it to include reviewing how we display manually created vs automatically created inbound returns 👍

**Pull request merged**